service/history/replication/raw_task_converter.go

```diff
@@ -653,13 +654,19 @@ func (c *syncVersionedTransitionTaskConverter) convert(
 		if taskInfo.FirstEventID == common.EmptyEventID && taskInfo.NextEventID == common.EmptyEventID && len(taskInfo.NewRunID) == 0 {
 			return nil, nil
 		}
+		releaseFunc(nil) // release wf lock before retrieving history events
```

Comment:
is lock required in generating backfill history task?

Reply:
No it is not required as mutable state is not used inside the func.